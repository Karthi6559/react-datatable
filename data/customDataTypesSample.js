import React from "react";
import { TableCell } from "@material-ui/core";

const customDataTypesSample = [
  {
    dataType: "number",
    component: cellVal => (
      <TableCell>
        <div style={{ color: "orange" }}>{cellVal}</div>
      </TableCell>
    )
  },
  {
    dataType: "text",
    component: cellVal => (
      <TableCell>
        <div style={{ color: "purple" }}>{cellVal}</div>
      </TableCell>
    )
  },
  {
    dataType: "iban",
    component: cellVal => (
      <TableCell>
        <div style={{ color: "red" }}>{cellVal}</div>
      </TableCell>
    )
  }
];

export default customDataTypesSample;
