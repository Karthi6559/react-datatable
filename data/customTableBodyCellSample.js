import React from "react";
import { TableCell } from "@material-ui/core";
import { cellValPropType, columnPropType } from "../src/proptypes";

const customTableBodyCellSample = ({ cellVal, column }) => {
  let val;
  switch (column.dataType) {
    case "boolean":
      if (cellVal) {
        val = (
          <div
            className="data-boolean"
            style={{ color: "green", textAlign: "center" }}
          >
            Yes
          </div>
        );
      } else {
        val = <div style={{ color: "red", textAlign: "center" }}>No</div>;
      }
      break;
    default:
      val = (
        <div className="data" style={{ color: "blue" }}>
          {cellVal}
        </div>
      );
      break;
  }
  if (val) {
    return <TableCell>{val}</TableCell>;
  }
  return <TableCell className="no-data" />;
};

customTableBodyCellSample.propTypes = {
  cellVal: cellValPropType,
  column: columnPropType
};

export default customTableBodyCellSample;
