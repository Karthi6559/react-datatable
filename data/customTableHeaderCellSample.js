import React from "react";
import { columnPropType } from "../src/proptypes";

const customTableHeaderCellSample = ({ column }) => {
  switch (column.dataType) {
    case "number":
      return <div className="number">{column.label}</div>;
    case "text":
      return <div className="text">{column.label}</div>;
    case "boolean":
      return <div className="boolean">{column.label}</div>;
    case "date":
      return <div className="date">{column.label}</div>;
    default:
      return <div className="default">{column.label}</div>;
  }
};

customTableHeaderCellSample.propTypes = {
  column: columnPropType
};

export default customTableHeaderCellSample;
