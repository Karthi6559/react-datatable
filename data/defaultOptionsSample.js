const defaultOptionsSample = {
  title: "",
  tableRef: null,
  dimensions: {
    datatable: {
      width: "100%",
      height: "100%",
      widthNumber: 0,
      totalWidthNumber: 0
    },
    header: {
      height: "0px",
      heightNumber: 0
    },
    body: {
      heightNumber: 0
    },
    row: {
      height: "60px",
      heightNumber: 0
    },
    columnSizeMultiplier: 1,
    isScrolling: false
  },
  keyColumn: null,
  font: "Roboto",
  data: {
    columns: [],
    rows: []
  },
  rowsEdited: [],
  rowsGlobalEdited: [],
  rowsSelected: [],
  newRows: [],
  rowsDeleted: [],
  actions: null,
  refreshRows: null,
  isRefreshing: false,
  stripped: false,
  searchTerm: "",
  orderBy: {
    keys: [],
    order: []
  },
  pagination: {
    pageSelected: 1,
    pageTotal: 1,
    rowsPerPageSelected: "",
    rowsCurrentPage: []
  },
  features: {
    canEdit: false,
    canAdd: false,
    canEditRow: null,
    canGlobalEdit: false,
    canPrint: false,
    canDownload: false,
    canDuplicate: false,
    canDelete: false,
    canSearch: false,
    canRefreshRows: false,
    canSelectRow: false,
    canOrderColumns: false,
    canSaveUserConfiguration: false,
    editableIdNewRow: [],
    userConfiguration: {
      columnsOrder: [],
      copyToClipboard: false
    },
    rowsPerPage: {
      available: [10, 25, 50, 100, "All"],
      selected: "All"
    },
    additionalActions: [],
    additionalIcons: [],
    selectionIcons: []
  }
};

export default defaultOptionsSample;
