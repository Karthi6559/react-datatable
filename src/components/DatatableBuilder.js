import React, { Component, Fragment } from "react";
import { Table } from "@material-ui/core";
import { connect } from "react-redux";
import Header from "./DatatableCore/Header";
import Body from "./DatatableCore/Body";
import { dataPropType } from "./proptypes/proptypes";

class DatatableBuilder extends Component {
  render() {
    const { data } = this.props;
    return (
      <Fragment>
        {data.columns.length > 0 && data.rows.length > 0 && (
          <Table>
            <Header />
            <Body />
          </Table>
        )}
        {(data.columns.length === 0 || data.rows.length === 0) && (
          <div id="no-rows">There is no data yet</div>
        )}
      </Fragment>
    );
  }
}

DatatableBuilder.propTypes = {
  data: dataPropType
};

const mapStateToProps = state => {
  return {
    data: state.datatableReducer.data
  };
};

export default connect(mapStateToProps)(DatatableBuilder);
