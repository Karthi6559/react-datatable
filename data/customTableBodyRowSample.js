import React from "react";
import {
  rowPropType,
  columnsOrderPropType,
  indexPropType,
  heightNumberPropType,
  columnSizeMultiplierPropType
} from "../src/proptypes";
import { simpleOptionsSample } from "./samples";

const customTableBodyRowSample = ({
  row,
  columnsOrder,
  rowIndex,
  columnSizeMultiplier,
  height
}) => {
  const { columns } = simpleOptionsSample.data;
  const columnAction = {
    id: "actions",
    label: "Actions",
    colSize: "150px",
    editable: false
  };
  return (
    <div
      className={
        rowIndex % 2 === 0 ? "Table-Row stripped" : "Table-Row not-stripped"
      }
      style={{
        height,
        background: rowIndex % 2 === 0 ? "#3f3f3f" : "#5f5f5f"
      }}
    >
      {columnsOrder.map(columnId => {
        const column =
          columnId === "actions"
            ? columnAction
            : columns.find(col => col.id === columnId);
        const width = `${(
          column.colSize.split("px")[0] * columnSizeMultiplier
        ).toString()}px`;
        return (
          <div className="Table-Cell" key={columnId}>
            <div style={{ color: "pink", width }}>{row[columnId]}</div>
          </div>
        );
      })}
    </div>
  );
};

customTableBodyRowSample.propTypes = {
  row: rowPropType,
  columnsOrder: columnsOrderPropType,
  rowIndex: indexPropType,
  height: heightNumberPropType,
  columnSizeMultiplier: columnSizeMultiplierPropType
};

export default customTableBodyRowSample;
