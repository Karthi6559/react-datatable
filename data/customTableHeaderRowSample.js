import React from "react";
import {
  columnsOrderPropType,
  columnSizeMultiplierPropType
} from "../src/proptypes";
import { simpleOptionsSample } from "./samples";

const customTableHeaderRowSample = ({ columnsOrder, columnSizeMultiplier }) => {
  const { columns } = simpleOptionsSample.data;
  const columnAction = {
    id: "o2xpActions",
    label: "Actions",
    colSize: "150px",
    editable: false
  };
  return (
    <div className="Table-Row">
      {columnsOrder.map(columnId => {
        const column =
          columnId === "o2xpActions"
            ? columnAction
            : columns.find(col => col.id === columnId);
        const width = `${(
          column.colSize.split("px")[0] * columnSizeMultiplier
        ).toString()}px`;
        return (
          <div className="Table-Header-Cell" key={column.id}>
            <div style={{ color: "green", width }}>{column.label}</div>
          </div>
        );
      })}
    </div>
  );
};

customTableHeaderRowSample.propTypes = {
  columnsOrder: columnsOrderPropType,
  columnSizeMultiplier: columnSizeMultiplierPropType
};

export default customTableHeaderRowSample;
