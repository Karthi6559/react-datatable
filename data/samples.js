import storeSample from "./storeSample";
import storeNoCustomComponentsSample from "./storeNoCustomComponentsSample";
import storeCustomTableBodyCellComponentSample from "./storeCustomTableBodyCellComponentSample";
import storeCustomTableBodyRowComponentSample from "./storeCustomTableBodyRowComponentSample";
import storeCustomTableHeaderCellComponentSample from "./storeCustomTableHeaderCellComponentSample";
import storeCustomTableHeaderRowComponentSample from "./storeCustomTableHeaderRowComponentSample";
import storeNoDataSample from "./storeNoDataSample";
import storeNoRowsDataSample from "./storeNoRowsDataSample";
import simpleOptionsSample from "./simpleOptionsSample";
import simpleOptionsNoDataSample from "./simpleOptionsNoDataSample";
import maximumOptionsSample from "./maximumOptionsSample";
import minimumOptionsSample from "./minimumOptionsSample";
import defaultOptionsSample from "./defaultOptionsSample";
import mergedSimpleOptionsSample from "./mergedSimpleOptionsSample";
import mergedMaximumOptionsSample from "./mergedMaximumOptionsSample";
import mergedMinimumOptionsSample from "./mergedMinimumOptionsSample";
import customTableBodyRowSample from "./customTableBodyRowSample";
import customTableBodyCellSample from "./customTableBodyCellSample";
import customTableHeaderRowSample from "./customTableHeaderRowSample";
import customTableHeaderCellSample from "./customTableHeaderCellSample";
import customDataTypesSample from "./customDataTypesSample";

export {
  storeSample,
  storeNoCustomComponentsSample,
  storeCustomTableBodyCellComponentSample,
  storeCustomTableBodyRowComponentSample,
  storeCustomTableHeaderCellComponentSample,
  storeCustomTableHeaderRowComponentSample,
  storeNoDataSample,
  storeNoRowsDataSample,
  simpleOptionsSample,
  simpleOptionsNoDataSample,
  maximumOptionsSample,
  minimumOptionsSample,
  defaultOptionsSample,
  mergedSimpleOptionsSample,
  mergedMaximumOptionsSample,
  mergedMinimumOptionsSample,
  customTableBodyRowSample,
  customTableBodyCellSample,
  customTableHeaderRowSample,
  customTableHeaderCellSample,
  customDataTypesSample
};
