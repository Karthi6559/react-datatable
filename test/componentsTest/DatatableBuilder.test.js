import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { storeSample, storeNoDataSample } from "../data/samples";
import DatatableBuilder from "../../src/components/DatatableBuilder";

describe("DatatableBuilder component", () => {
  const mockStore = configureStore();
  beforeEach(() => {});
  it("should render without errors", () => {
    const store = mockStore(storeSample);
    const tree = renderer.create(
      <Provider store={store}>
        <DatatableBuilder />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });

  it("should render without errors", () => {
    const store = mockStore(storeNoDataSample);
    const tree = renderer.create(
      <Provider store={store}>
        <DatatableBuilder />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
