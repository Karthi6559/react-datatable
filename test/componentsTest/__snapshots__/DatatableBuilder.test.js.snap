// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatatableBuilder component should render without errors 1`] = `
<table
  className="MuiTable-root-1"
>
  <thead
    className="MuiTableHead-root-2"
  >
    <tr
      className="MuiTableRow-root-3 MuiTableRow-head-6"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9 MuiTableCell-alignLeft-16"
        scope="col"
      >
        <div>
          id
        </div>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9 MuiTableCell-alignRight-18"
        scope="col"
      >
        <div>
          name
        </div>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9 MuiTableCell-alignRight-18"
        scope="col"
      >
        <div>
          age
        </div>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9 MuiTableCell-alignRight-18"
        scope="col"
      >
        <div>
          adult
        </div>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9 MuiTableCell-alignRight-18"
        scope="col"
      >
        <div>
          birth date
        </div>
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-20"
  >
    <tr
      className="MuiTableRow-root-3"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignLeft-16"
        scope="row"
      >
        <div>
          50cf
        </div>
      </th>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div>
          28
        </div>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div>
          Kerr Mayo
        </div>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div />
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div>
          1972-09-04T11:09:59
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root-3"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignLeft-16"
        scope="row"
      >
        <div>
          209
        </div>
      </th>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div>
          34
        </div>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div>
          Freda Bowman
        </div>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div />
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div>
          1988-03-14T09:03:19
        </div>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root-3"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignLeft-16"
        scope="row"
      >
        <div>
          2dd81ef
        </div>
      </th>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div>
          14
        </div>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div>
          Becky Lawrence
        </div>
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div />
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10 MuiTableCell-alignRight-18"
      >
        <div>
          1969-02-10T04:02:44
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`DatatableBuilder component should render without errors 2`] = `
<div
  id="no-rows"
>
  There is no data yet
</div>
`;
